### PR TITLE
Only offer MoveType if the user is on the header for a type.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/15980

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest this please

File: src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.cs [hunk 1]
```diff
@@ -29,32 +31,35 @@ private enum OperationKind
...
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            if (!ShouldAnalyze(root, textSpan))
-            {
-                return default(ImmutableArray<CodeAction>);
-            }
-
-            var semanticDocument = await SemanticDocument.CreateAsync(document, cancellationToken).ConfigureAwait(false);
-            var state = State.Generate((TService)this, semanticDocument, textSpan, cancellationToken);
-            if (state == null)
+            if (textSpan.IsEmpty)
```
Comment (Pilchie) [inline src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.cs]: Why the change to only do it on empty selections?  Many people will double click a name and expect something like this to work.

File: src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.cs [hunk 2]
```diff
@@ -29,32 +31,35 @@ private enum OperationKind
...
+                var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var nodeToAnalyze = root.FindToken(textSpan.Start).GetAncestor<TTypeDeclarationSyntax>();
+                if (nodeToAnalyze != null)
+                {
+                    var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+                    if (syntaxFacts.IsOnTypeHeader(root, textSpan.Start))
+                    {
+                        var semanticDocument = await SemanticDocument.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+                        var state = State.Generate(
+                            semanticDocument, textSpan, nodeToAnalyze, cancellationToken);
```
Comment (Pilchie) [inline src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.cs]: If we *do* only allow positions, why not change this to `int position` instead of `textSpan`.
